mds/apis/prv_api/vehicles.py

```diff
@@ -68,9 +68,7 @@ class DeviceSerializer(serializers.ModelSerializer):
         allow_null=True,
     )
     battery = serializers.FloatField(
-        source="latest_event.properties.telemetry.battery_pct",
-        help_text="Percentage between 0 and 1",
-        allow_null=True,
+        source="dn_battery_pct", help_text="Percentage between 0 and 1", allow_null=True
```

Comment:
Seems fine but not related to this PR.

... Or is it? If you change just that, do we need to fetch `latest_event` in current state at all?

Reply:
It is, actually, there is two ways to fix the slowness:
- change the latest_events query that is taking too long because we have too many evenrecords (fix done). This query is also used in agency API
- as we have dn_battery_pct in device model, we might as well use this value so we can avoid doing queries.